src/sentry/coreapi.py

```diff
@@ -1085,6 +1080,12 @@ def _decode(self):
 
         data['sdk']['client_ip'] = self._client_ip
 
+        if 'project' not in data:
+            data['project'] = self._project.id
+
+        if 'key_id' not in data:
```

Comment:
This shouldn't be conditionally setting. We can't allow the client to send these values and override our fixed values. That means someone sending in their own key_id will be problematic.